quickwit-metastore/src/metastore.rs

```diff
@@ -108,22 +108,37 @@ pub struct MetadataSet {
 #[async_trait]
 pub trait Metastore: Send + Sync + 'static {
     /// Creates an index.
+    /// This API creates index metadata set in the metastore.
+    /// An error will occur if an index that exists in the storage is specified.
     async fn create_index(&self, index_id: &str, doc_mapping: DocMapping) -> MetastoreResult<()>;
 
     /// Deletes an index.
+    /// This API removes the specified index metadata set from the metastore,
+    /// but does not remove the index from the storage.
+    /// An error will occur if an index that does not exist in the storage is specified.
     async fn delete_index(&self, index_id: &str) -> MetastoreResult<()>;
 
     /// Stages a split.
+    /// A split needs to be Staged before uploading any of its files to the storage.
+    /// An error will occur if an index that does not exist in the storage is specified.
+    /// Also, an error will occur if you specify a split that already exists.
     async fn stage_split(
         &self,
         index_id: &str,
         split_metadata: SplitMetadata,
     ) -> MetastoreResult<()>;
 
     /// Publishes a split.
+    /// This API only updates the state of the split from Staged to Published.
+    /// At this point, the split files are assumed to have already been uploaded.
+    /// If the split is already published, this API call returns a success.
+    /// An error will occur if you specify an index or split that does not exist in the storage.
     async fn publish_split(&self, index_id: &str, split_id: &str) -> MetastoreResult<()>;
 
     /// Lists the splits.
+    /// Returns a list of splits that intersect the given time_range and split_state.
+    /// Regardless of the time range filter, if a split has no timestamp it is always returned.
```

Comment:
The current implementation in the single file store is different.

We need to change the doc or change the code and the unit tests.

Reply:
The documentation is wrong and will be corrected.

Reply:
I misread the specification, so I will fix the code and add some tests.

Reply:
Fixed a bug in the time range specification and added the unit tests.